pkg/k8shandler/visualization.go

```diff
@@ -236,20 +241,26 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateOauthClient(oauthSecr
 				return fmt.Errorf("Failed to get %v oauthclient for %q: %v", oauthClient.Name, clusterRequest.cluster.Name, err)
 			}
 
+			//Check to see if we even need to update
+			if areOAuthClientsSame(current, oauthClient) {
+				return nil
+			}
+
+			updatedAt = time.Now().String()
 			current.RedirectURIs = oauthClient.RedirectURIs
 			current.Secret = oauthClient.Secret
 			current.ScopeRestrictions = oauthClient.ScopeRestrictions
+			current.GetObjectMeta().GetAnnotations()[annotationOauthSecretUpdatedAt] = updatedAt
```

Comment:
Just got a crash at this line:
```
E0730 22:20:34.279519       1 runtime.go:69] Observed a panic: "assignment to entry in nil map" (assignment to entry in nil map)
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:573
/usr/local/go/src/runtime/panic.go:502
/usr/local/go/src/runtime/hashmap_fast.go:696
/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/visualization.go:253
```
The rest of the stacktrace is included below but the crash is at this line - either `current.GetObjectMeta()` or `current.GetObjectMeta().GetAnnotations()` is `nil`

```
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/client-go/util/retry/util.go:64
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:203
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/client-go/util/retry/util.go:63
/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/visualization.go:234
/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/visualization.go:60
/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/reconciler.go:28
/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/controller/clusterlogging/clusterlogging_controller.go:100
/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215
/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2361
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 507 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x13b83c0, 0x16b0520)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/openshift/cluster-logging-operator/pkg/k8shandler.(*ClusterLoggingRequest).createOrUpdateOauthClient.func1(0x1f2c520, 0x411b09)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/visualization.go:253 +0x36f
k8s.io/client-go/util/retry.RetryOnConflict.func1(0xc420c49888, 0x412358, 0x20)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/client-go/util/retry/util.go:64 +0x36
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0xc420e2a740, 0x412358, 0x30)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:203 +0x9c
k8s.io/client-go/util/retry.RetryOnConflict(0x989680, 0x3ff0000000000000, 0x3fb999999999999a, 0x5, 0xc420dca4b0, 0xe, 0xc42059f460)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/client-go/util/retry/util.go:63 +0xba
github.com/openshift/cluster-logging-operator/pkg/k8shandler.(*ClusterLoggingRequest).createOrUpdateOauthClient(0xc42070ed00, 0xc420043280, 0x40, 0x40, 0xc420043280, 0x40, 0x0)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/visualization.go:234 +0x656
github.com/openshift/cluster-logging-operator/pkg/k8shandler.(*ClusterLoggingRequest).CreateOrUpdateVisualization(0xc42070ed00, 0x0, 0x0)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/visualization.go:60 +0x1f0
github.com/openshift/cluster-logging-operator/pkg/k8shandler.Reconcile(0xc420a3a000, 0x16da3e0, 0xc4206e3bf0, 0xc420789fe0, 0x11)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/k8shandler/reconciler.go:28 +0x27b
github.com/openshift/cluster-logging-operator/pkg/controller/clusterlogging.(*ReconcileClusterLogging).Reconcile(0xc420704560, 0xc420789fe0, 0x11, 0xc4208fde50, 0x8, 0x1f2b180, 0x3, 0x0, 0x3)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/openshift/cluster-logging-operator/pkg/controller/clusterlogging/clusterlogging_controller.go:100 +0x167
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc4209705a0, 0x0)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x188
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4206fe3b0)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4206fe3b0, 0x3b9aca00, 0x0, 0x1, 0xc42099aa20)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
k8s.io/apimachinery/pkg/util/wait.Until(0xc4206fe3b0, 0x3b9aca00, 0xc42099aa20)
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x35b
```